CHANGELOG.md

```diff
@@ -38,7 +38,7 @@ mapView.location2.puckBearingEnabled = false
 * Fix skipping / crashing user events scheduled on a render thread with `MapView#queueEvent`. ([#1068](https://github.com/mapbox/mapbox-maps-android/pull/1068))
 * Fix location puck not being shown if map is created without initial style (e.g. MapInitOptions.styleUri == null) and then loaded asynchronously. ([#1114](https://github.com/mapbox/mapbox-maps-android/pull/1114))
 * Fix crash within location plugin that happens when style is reloaded simultaneously with location plugin updates. ([#1112](https://github.com/mapbox/mapbox-maps-android/pull/1112))
-* Fix memory leak in location component. ([#1093](https://github.com/mapbox/mapbox-maps-android/pull/1093))
+* Fix memory leak in location component. ([#1093](https://github.com/mapbox/mapbox-maps-android/pull/1093), [#1172](https://github.com/mapbox/mapbox-maps-android/pull/1172))
```

Comment:
as I've said before I'd remove link to #1093 as it does not actually fix the leak if I understood correctly

Reply:
Actually #1093 resolved it, by adding WeakReference context that lives on CompassEngine is no more kept by GoogleLocationEngine callback.  

And 1172 just adds more safety, additionally using applicationContext everywhere in the LocationEngine/CompassEngine, thus making any leaks not possible now and in future.

Basically any of those fixes would be sufficient, and both of them work too =)